test/test_transformers.py

```diff
@@ -539,6 +539,45 @@ def test_padding_and_src_mask_bool(self):
             src_key_padding_mask=padding_mask,
         )
 
+    def test_decoder_padding_and_src_mask_bool(self):
```

Comment:
What is this testing? We can use `self.assertEqual` and such to compare outputs.

Reply:
[`assertNoLogs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertNoLogs) might be able to turn this into a test that fails if we do start to log again.

Reply:
Only introduced in 3.11 :(